config/config.md

```diff
@@ -382,7 +382,8 @@
 
 | Key | Type | Default | Descriptions |
 | --- | -----| ------- | ----------- |
-| `node_id` | Integer | Unset | The datanode identifier and should be unique in the cluster. |
+| `node_id` | Integer | Unset | -- |
+| `node_mode` | String | `read_write` | The mode of the datanode.<br/>Available values: `read_write`, `read_only`, `write_only`.<br/><br/>The default value is `read_write`, which can handle both writable and read-only regions.<br/>- When the mode is `read_only`, only read-only regions will be assigned to this node.<br/>- When the mode is `write_only`, only writable regions will be assigned to this node. |
```

Comment:
Instread `node_mode`, I prefer `node_role` personally:

* `Default` for read and write,
* `Ingester` for write only,
* `Querier` for read only,
* `Compactor` for comaction tasks,
* `Indexer` for indexing

What do you think? @WenyXu @fengjiachun

Reply:
A better approach is to define roles as flags, allowing a node to assume multiple roles. For example: `node_role: Compactor | Indexer`.

Reply:
1. I like the idea of `node_mode` (Default, Ingester, Querier, etc.). What do you think, @WenyXu?
2. We use `node_mode` since the `role` is already taken. @killme2008 
```
role
 - datanode
   - node_mode
     - Default
     - Ingester
     - ...
 - flownode
 - frontend
   - node_mode
     ..
```

Reply:
The field name `role` is already taken in the protocol. How about adding a new field like `role_details` instead?

```
message RequestHeader {
  Role role = 1;
  oneof role_details {
    DatanodeDetails datanode = 2;
    FrontendDetails frontend = 3;
    // future...
  }
}

message DatanodeDetails {
  repeated DatanodeTask tasks = 1;
}

enum DatanodeTask {
  DEFAULT = 0;
  INGESTER = 1;
  QUERIER = 2;
  INDEXER = 3;
  COMPACTOR = 4;
}
```

Reply:
How about `workload_types`

For datanode it can be:

- hybrid (default)
- ingest
- query
- compaction
- index

For frontend, in future there will be

- default
- streaming
- bulk
- query

Reply:
Or workload_mode

Reply:
The `workload_types` looks good to me. The proto will look like this:

```
message RequestHeader {
  Role role = 1;
  oneof workload_types {
    DatanodeWorkloads datanode = 2;
    DatanodeWorkloads frontend = 3;
    // future...
  }
}

message DatanodeWorkloads {
  repeated DatanodeWorkload workloads = 1;
}

enum DatanodeWorkload {
  DEFAULT = 0;
  INGEST = 1;
  QUERY = 2;
  INDEX = 3;
  COMPACTION = 4;
}
```

Reply:
`workload_type` seems appropriate, The keypoint is that Datanodes must be able to handle mixed types.